File: web/src/features/team/types/worker-profile.ts
export type WorkerProfile = {
  id: string;
  name: string;
  email: string;
  role: string;
  color: string;
  skills: number[];
  available: boolean;
  capacityMinutes: number;
};

export type WorkerProfileRead = {
  revision: number;
  profile: WorkerProfile | null;
};

File: mobile/lib/features/profile/domain/worker_profile.dart
import '../../../core/data/json.dart';

class WorkerProfile {
  WorkerProfile.fromJson(Json data) : data = Map.unmodifiable(data);
  final Json data;
  String get name => string(data['name']);
  String get email => string(data['email']);
  String get role => string(data['role']);
  bool get available => data['available'] == true;
  List<int> get skills => (data['skills'] as List? ?? [])
      .whereType<num>()
      .map((v) => v.toInt())
      .toList();
  int get capacity => integer(data['capacityMinutes']);
}
